scripts/install_environment.sh

```diff
@@ -21,9 +21,43 @@ function install_libvirt() {
         libvirt-devel \
         libvirt-daemon-kvm \
         qemu-kvm \
-        libgcrypt \
-        swtpm \
-        swtpm-tools
+        libgcrypt
+
+    # TODO: get swtpm RPMs from EPEL as soon as 0.7.1 or later lands there.
+    # based on: https://github.com/stefanberger/swtpm/wiki#compile-and-install-on-linux
+
+    echo "Enable EPEL"
+    dnf install -y  epel-release
+
+    echo "Install swtpm dependencies"
+    sudo dnf install -y \
+        libtasn1-devel \
+        expect \
+        socat \
+        python3-twisted \
+        fuse-devel \
+        glib2-devel \
+        gnutls-devel \
+        gnutls-utils \
+        gnutls \
+        json-glib-devel \
+        autoconf \
+        automake \
+        libtool \
+        openssl-devel \
+        libtpms \
+        libtpms-devel \
+        libseccomp-devel \
+        net-tools \
+        iproute \
+        git \
+        make
+
+    echo "Install swtpm v0.7.1 from source"
+    git clone --depth 1 --branch v0.7.1 https://github.com/stefanberger/swtpm.git
+    pushd swtpm
+    ./autogen.sh --with-openssl --prefix=/usr
```

Comment:
don't we need `make` + `make install` as described in the doc?

also, I see that `swtmp` packages are still installed in the build log: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_assisted-test-infra/1748/pull-ci-openshift-assisted-test-infra-master-e2e-metal-assisted/1547106888800276480/artifacts/e2e-metal-assisted/assisted-baremetal-setup/build-log.txt

Reply:
how is that possible?

Reply:
Added those.
as for why the RPMs are still installed - I'm not sure. I don't see any other places in the code that mention those RPMs.
@osherdp @eliorerz , do you know?

Reply:
So as @adriengentil mentioned over Slack, swtpm gets installed as a dependency of [libvirt-daemon-driver-qemu](https://pkgs.org/download/libvirt-daemon-driver-qemu).

I have tested `install_environment.sh` locally on a RockyLinux 8.6 container, and the install from source procedure seems to override the older swtpm binaries:


**check done after swtpm-0.7.0 got installed via RPM**
```sh
[root@5091c205cc54 /]# swtpm --version
TPM emulator version 0.7.0, Copyright (c) 2014-2021 IBM Corp.
[root@5091c205cc54 /]# which swtpm
/usr/bin/swtpm


[root@5091c205cc54 /]# /usr/bin/swtpm_ioctl --version
TPM emulator control tool version 0.7.0, Copyright (c) 2015 IBM Corp.
```

**a check done after swtpm-0.7.1 got installed from source**

```sh
[root@5091c205cc54 /]# /usr/bin/swtpm_ioctl --version
TPM emulator control tool version 0.7.1, Copyright (c) 2015 IBM Corp.
[root@5091c205cc54 /]# swtpm --version
TPM emulator version 0.7.1, Copyright (c) 2014-2021 IBM Corp
```

can we continue with that approach?